dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/HTMLPageAssetRenderedAPIImpl.java

```diff
@@ -256,6 +292,322 @@ public ViewAsPageStatus getViewAsStatus(final HttpServletRequest request,
+                    }
+                }
+
+                contentMaps.put(CONTAINER_UUID_PREFIX + uniqueUUIDForRender, personalizedContentletMap);
+            }
+
+//            for (final Map.Entry<String, List<String>> entry : containerUuidPersona.entrySet()) {
+//
+//                contextMap.put("contentletList" + entry.getKey(), entry.getValue());
+//                contextMap.put("totalSize"      + entry.getKey(), entry.getValue().size());
+//            }
+
+            raws.add(new ContainerRaw(container, containerStructures, contentMaps));
+        }
+
+        return raws;
+    }
+
+    private boolean isLive(final HttpServletRequest request, final PageMode mode) {
+        return (request == null || request.getSession(false) == null
+                || request.getSession().getAttribute("tm_date") == null) && mode.showLive;
+    }
+
+    private String getPersonaTagToIncludeContent(final HttpServletRequest request, final Set<String> personalizationsForPage) {
+        IPersona iPersona = null;
+
+        if (request != null) {
+            final Optional<Visitor> visitor = APILocator.getVisitorAPI().getVisitor(request);
+            iPersona = visitor.isPresent() && visitor.get().getPersona() != null ? visitor.get().getPersona() : null;
+        }
+
+        final String currentPersonaTag = iPersona == null ? MultiTree.DOT_PERSONALIZATION_DEFAULT
+                : Persona.DOT_PERSONA_PREFIX_SCHEME + StringPool.COLON + iPersona.getKeyTag();
+
+        final boolean hasPersonalizations = personalizationsForPage.contains(currentPersonaTag);
+
+        return hasPersonalizations ? currentPersonaTag : MultiTree.DOT_PERSONALIZATION_DEFAULT;
+    }
+
+    private Container getContainer(final boolean live, final String containerId, final Host site) throws DotSecurityException, DotDataException {
+        Container container;
+        final WorkingContainerFinderByIdOrPathStrategyResolver strategyResolver =
+                WorkingContainerFinderByIdOrPathStrategyResolver.getInstance();
+        final Optional<ContainerFinderByIdOrPathStrategy> strategy = strategyResolver.get(containerId);
+        final ContainerFinderByIdOrPathStrategy workingStrategy = strategy.isPresent() ? strategy.get() : strategyResolver.getDefaultStrategy();
+        final Supplier<Host> resourceHostSupplier = () -> site;
+
+        try {
+            if (live) {
+
+                container = this.getLiveContainerById(containerId, site);
+                if (null == container) {
+                    container = workingStrategy.apply(containerId, APILocator.systemUser(), false, resourceHostSupplier);
+                }
+            } else {
+                container = workingStrategy.apply(containerId, APILocator.systemUser(), false, resourceHostSupplier);
+            }
+        } catch (NotFoundInDbException | DotRuntimeException e) {
+
+            new ContainerExceptionNotifier(e, containerId).notifyUser();
+            container = null;
```

Comment:
Codacy found an issue: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6090726)